libcaf_core/caf/intrusive_ptr.hpp

```diff
@@ -231,45 +231,45 @@ bool operator!=(std::nullptr_t, const intrusive_ptr<T>& x) {
 
 /// @relates intrusive_ptr
 template <class T, class U>
-std::enable_if_t<detail::is_comparable<T*, U*>::value, bool>
-operator==(const intrusive_ptr<T>& lhs, const U* rhs) {
+  requires detail::is_comparable<T*, U*>::value
```

Comment:
@riemass FYI, I'm already working on a followup that will clean places where we still use `::value`.